pkg/astnormalization/implicit_extend_root_operation.go

```diff
@@ -0,0 +1,57 @@
+package astnormalization
+
+import (
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astvisitor"
+)
+
+/*
+type Query {...}
+type Mutation {...}
+type Subscription {...}
+
+will be,
+
+extend type Query {...}
+extend type Mutation {...}
+extend type Subscription {...}
+
+this also works if root types are defined in schema{...} with other names.
+root types are left unmodified if they have no fields, directives or implements any interface.
+*/
+type implicitExtendRootOperationVisitor struct {
+	operation *ast.Document
+}
+
+func implicitExtendRootOperation(walker *astvisitor.Walker) {
+	v := &implicitExtendRootOperationVisitor{}
+	walker.RegisterEnterDocumentVisitor(v)
+	walker.RegisterEnterObjectTypeDefinitionVisitor(v)
+}
+
+func (v *implicitExtendRootOperationVisitor) EnterDocument(operation, _ *ast.Document) {
+	v.operation = operation
+}
+
+func (v *implicitExtendRootOperationVisitor) EnterObjectTypeDefinition(ref int) {
+	node := v.operation.ObjectTypeDefinitions[ref]
+	if !(node.HasDirectives || node.HasFieldDefinitions || len(node.ImplementsInterfaces.Refs) > 0) {
+		return
+	}
+	switch v.operation.Input.ByteSliceString(node.Name) {
+	case implicitQueryTypeName, implicitMutationTypeName, implicitSubscriptionTypeName,
```

Comment:
probably you could omit here implicit names, as u will have them in index any way
and with bytes helper, you could use `Equals` of `ByteSlice` to not convert values to strings

smth like this
```go
if v.operation.Index.QueryTypeName.Equals(v.operation.ObjectTypeDefinitionNameBytes(ref)) ||
  v.operation.Index.MutationTypeName.Equals(v.operation.ObjectTypeDefinitionNameBytes(ref)) ||
v.operation.Index.SubscriptionTypeName.Equals(v.operation.ObjectTypeDefinitionNameBytes(ref)) {
  // logic here
}
```

Reply:
I think the current semantics is more readable